File: GoLang/剑指Offer/easy/剑指Offer_27_二叉树的镜像.go
package easy

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func mirrorTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}
	helper27(root)
	return root
}

func helper27(root *TreeNode) {
	if root == nil {
		return
	}

	if root.Left != nil || root.Right != nil {
		temp := root.Left
		root.Left = root.Right
		root.Right = temp

		helper27(root.Left)
		helper27(root.Right)
	}
}

File: Java/剑指Offer/easy/剑指Offer_27_二叉树的镜像.java
package easy;

import javax.swing.tree.TreeNode;

/**
 * Created by lwd at 2021/7/9
 *
 * @Description:
 */

public class 剑指Offer_27_二叉树的镜像 {
    public TreeNode mirrorTree(TreeNode root) {
        if (root == null) return null;

        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;

        mirrorTree(root.left);
        mirrorTree(root.right);
        return root;
    }

    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

}
